src/main/resources/+ciplugins/+jenkins/BuildReportPlugin.m

```diff
@@ -6,7 +6,13 @@
 
         function runTaskGraph(plugin, pluginData)
             runTaskGraph@matlab.buildtool.plugins.BuildRunnerPlugin(plugin, pluginData);
-            fID = fopen(fullfile(getenv("WORKSPACE"),'.matlab/buildArtifact.json'), 'w');
+            [fID, msg] = fopen(fullfile(getenv("WORKSPACE"),'.matlab/buildArtifact.json'), 'w');
+
+            if fID == -1
+                warning('BuildTool:artifactFileWarnning','Could not open a file for Jenkins build result table due to: %s', msg);
+                fID = 1;
```

Comment:
Arbitrarily setting the fID would throw an error while closing the file using 'fclose'. 
![image](https://github.com/mathworks/jenkins-matlab-plugin/assets/70003902/64cbe91b-0061-4098-b208-6bb73be2a112)

Reply:
Its good point Vahila, may be we can keep the entire fprintf in else part